cpp/src/parquet/thrift_internal.h

```diff
@@ -361,6 +361,9 @@ inline void DeserializeThriftUnencryptedMsg(const uint8_t* buf, uint32_t* len,
   shared_ptr<ThriftBuffer> tmem_transport(
       new ThriftBuffer(const_cast<uint8_t*>(buf), *len));
   apache::thrift::protocol::TCompactProtocolFactoryT<ThriftBuffer> tproto_factory;
+  // Protect against CPU and memory bombs
+  tproto_factory.setStringSizeLimit(10 * 1000 * 1000);
+  tproto_factory.setContainerSizeLimit(10 * 1000 * 1000);
```

Comment:
@wesm I need validation on this.

Reply:
Note that these are the limits for individual strings and containers in thrift-encoded data, not a cumulated limit.

Reply:
I think this is fine. We recently set string statistics to not write if the exceed 4K length. There may be cases where people have files with statistics more than 10MB, but I would rather hear from them and we can see how to add public APIs to allow for very large metadata.